embedded-graphics/src/primitives/rectangle.rs

```diff
@@ -60,20 +61,35 @@ impl Dimensions for Rectangle {
     }
 
     fn bottom_right(&self) -> Point {
-        self.bottom_right
+        if self.size == Size::zero() {
+            // FIXME: The bottom right corner isn't valid if the size of the rectangle is zero,
+            //        because creating a rectangle from two corner points creates a rectangle that
+            //        is at least 1x1 pixels large.
+            self.top_left
+        } else {
+            self.top_left + self.size - Point::new(1, 1)
+        }
     }
 
     fn size(&self) -> Size {
-        Size::from_bounding_box(self.top_left, self.bottom_right)
+        self.size
     }
 }
 
 impl Rectangle {
-    /// Create a new rectangle from the top left point to the bottom right point with a given style
-    pub const fn new(top_left: Point, bottom_right: Point) -> Self {
+    /// Creates a new rectangle from the top left point and the size.
+    pub const fn new(top_left: Point, size: Size) -> Self {
+        Rectangle { top_left, size }
+    }
+
+    /// Creates a new rectangle from two corners.
+    pub fn with_corners(corner_1: Point, corner_2: Point) -> Self {
```

Comment:
```suggestion
    pub fn from_corners(corner_1: Point, corner_2: Point) -> Self {
```

Fits Rust API guidelines better, and we already have stuff like `Triangle::from_points`.

Reply:
Does it? I guess this is a case that's not 100% defined. This were my thoughts when choosing this name:
`from_...` is a conversion constructor and IMO two points can't be converted into a rectangle, but  a rectangle can be constructed by specifying two of it's corners. And as a general constructor with additional details it's supposed to use a `with_` prefix.

Reply:
Could be argued either way I spose. Let's leave it as `with_corners` then :). Once the changelog item is added I'll approve the PR. Needs (another) rebase too 😰

Reply:
I did just another look at this and maybe we should find a more consistent naming scheme.

`Triangle::from_points` is different than this, because it uses the same points to define a triangle as `Triangle::new` but as an array instead of three arguments. So maybe `Triangle::from_points` shouldn't have been named this way.

Suggestion:
* Use `vertex` instead of `corner`
* Change `Triangle` to also use `vertex` instead of the generic `points`.
  This would also make it possible to use `points()` for the iterator suggested in #156, with a lower risk of confusion between vertices and points.

Reply:
If I understand correctly:

* Change `Triangle::from_points` to `Triangle::from_vertices`. This continues to take a list of 3 points. This might also wanna be `Triangle::with_vertices` if we're going for consistency?
* Change `Rectangle::with_corners` to `Rectangle::with_vertices`. Does this remain as two arguments or should we change it to take a list to be consistent with `Triangle`?

?

Reply:
> If I understand correctly:

I'm not quite sure myself and open to better suggestions.

> * Change `Triangle::from_points` to `Triangle::from_vertices`. This continues to take a list of 3 points. This might also wanna be `Triangle::with_vertices` if we're going for consistency?

Maybe we should make `Triangle` behave consistently with `Polygon`. This would probably mean that `Triangle::new` and `Poygon::new` would have a slice as an argument. `Triangle::with_vertices` could then be used for specifying three individual points.

> Change `Rectangle::with_corners` to `Rectangle::with_vertices`. Does this remain as two arguments or should we change it to take a list to be consistent with `Triangle`?

If we change `Triangle` like in the above comment `Rectangle::with_vertices` could still use two arguments. Creating a rectangle from a slice of two points isn't a common operation IMO.